pkg/network/ambassador.go

```diff
@@ -0,0 +1,81 @@
+package network
+
+import (
+	"github.com/labstack/gommon/log"
+	crypto "github.com/nuts-foundation/nuts-crypto/pkg"
+	"github.com/nuts-foundation/nuts-crypto/pkg/cert"
+	network "github.com/nuts-foundation/nuts-network/pkg"
+	"github.com/nuts-foundation/nuts-network/pkg/model"
+	"github.com/nuts-foundation/nuts-registry/pkg/events"
+	"github.com/nuts-foundation/nuts-registry/pkg/events/domain"
+	errors2 "github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+const documentType = "nuts.registry-event"
+
+// Ambassador acts as integration point between the registry and network by sending registry events to the
+// network and (later on) process notifications of new documents on the network that might be of interest to the registyr.
+type Ambassador interface {
+	RegisterEventHandlers(fn events.EventRegistrar, eventType []events.EventType)
+}
+
+type ambassador struct {
+	networkClient network.NetworkClient
+	cryptoClient  crypto.Client
+}
+
+func NewAmbassador(networkClient network.NetworkClient, cryptoClient crypto.Client) Ambassador {
+	return &ambassador{
+		networkClient: networkClient,
+		cryptoClient:  cryptoClient,
+	}
+}
+
+func (n *ambassador) RegisterEventHandlers(fn events.EventRegistrar, eventType []events.EventType) {
+	for _, eventType := range eventType {
+		fn(eventType, func(event events.Event, lookup events.EventLookup) error {
+			if event.Type() == domain.RegisterVendor {
+				if err := n.registerVendorCertificateInTrustStore(event); err != nil {
+					logrus.Errorf("Error while registering vendor certificates in the truststore (event=%s): %v", event.Ref(), err)
+				}
+			}
+			go n.sendEventToNetwork(event)
```

Comment:
aren't events also generated from the network? eg: infinite loop?

Reply:
Yes they're generated from the network but before they're published (by this code) it checks whether it already exists. In fact there's a TODO that a vendor should only publish its own events, so I'll fix that.